records are missing when using rewrite tag to emit one record to 2 different destinations.

## Bug Report
I have a container in k8s which emitting logs like:

          i=0;
          while true;
          do
            echo "[DEBUG] i=$i"
            echo "[ERROR] i=$i"
            i=$((i+1));
            sleep 1;
          done
the fluent-bit filters and outputs are
```
 [FILTER]
        Name modify
        match  kube.var.log.containers*_log-1_*
        Remove $kubernetes
    [FILTER]
        Name modify
        match  kube.var.log.containers*_log-1_*
        Add test1 true
        Condition Key_value_matches message ERROR
    [FILTER]
        Name modify
        match  kube.var.log.containers*_log-1_*
        Add test2 true
        Condition Key_value_matches message DEBUG
    [FILTER]
        Name modify
        match  kube.var.log.containers*_log-1_*
        Add test3 true
        Condition Key_value_matches message DEBUG
    [FILTER]
        Name rewrite_tag
        Match kube.var.log.containers*_log-1_*
        Rule $test1  ^(true)$ test1 true
        Emitter_Mem_Buf_Limit  10M
        Emitter_Storage.type memory
    [FILTER]
        Name rewrite_tag
        Match kube.var.log.containers*_log-1_*
        Rule $test2  ^(true)$ test2 true
        Emitter_Mem_Buf_Limit  10M
        Emitter_Storage.type memory
    [FILTER]
        Name rewrite_tag
        Match kube.var.log.containers*_log-1_*
        Rule $test3  ^(true)$ test3 true
        Emitter_Mem_Buf_Limit  10M
        Emitter_Storage.type memory
    [OUTPUT]
        Name file
        Match  test1
        File /tmp/test1
    [OUTPUT]
        Name file
        Match  test2
        File /tmp/test2
    [OUTPUT]
        Name file
        Match  test3
        File /tmp/test3
    [OUTPUT]
        Name file
        Match kube*
        File /tmp/kube
```
I want to collect error logs in /tmp/test1, and collect debug logs both in /tmp/test2 and /tmp/test3.
I found the log num in test1 has no problem. but in /tmp/test2 and /tmp/test3 the num is not continuous.
fluent-bit version is 1.8.10
```
test3: [1640163187.297961204, {"stream":"stdout","logtag":"F","message":"[DEBUG] i=581","kubernetes":{"pod_name":"counter-1","namespace_name":"log-1","pod_id":"d76c604a-dc33-4ab6-b1b5-0f3bad72555d","host":"5c132468-b25d-4dbf-b71e-4a4aee46a011","container_name":"container1","docker_id":"f3be888c4a6d5bd71e1c00a7de63fe0bbc73df279634ca207ab6bb3e73d2d7b2","container_hash":"gcr.io/cf-pks-releng-environments/busybox@sha256:679b1c1058c1f2dc59a3ee70eed986a88811c0205c8ceea57cec5f22d2c3fbb1","container_image":"gcr.io/cf-pks-releng-environments/busybox:latest"},"test2":"true","test3":"true","cluster_name":"aaa"}]
test3: [1640163203.320908254, {"stream":"stdout","logtag":"F","message":"[DEBUG] i=597","kubernetes":{"pod_name":"counter-1","namespace_name":"log-1","pod_id":"d76c604a-dc33-4ab6-b1b5-0f3bad72555d","host":"5c132468-b25d-4dbf-b71e-4a4aee46a011","container_name":"container1","docker_id":"f3be888c4a6d5bd71e1c00a7de63fe0bbc73df279634ca207ab6bb3e73d2d7b2","container_hash":"gcr.io/cf-pks-releng-environments/busybox@sha256:679b1c1058c1f2dc59a3ee70eed986a88811c0205c8ceea57cec5f22d2c3fbb1","container_image":"gcr.io/cf-pks-releng-environments/busybox:latest"},"test2":"true","test3":"true","cluster_name":"aaa"}]
test3: [1640163213.331465903, {"stream":"stdout","logtag":"F","message":"[DEBUG] i=607","kubernetes":{"pod_name":"counter-1","namespace_name":"log-1","pod_id":"d76c604a-dc33-4ab6-b1b5-0f3bad72555d","host":"5c132468-b25d-4dbf-b71e-4a4aee46a011","container_name":"container1","docker_id":"f3be888c4a6d5bd71e1c00a7de63fe0bbc73df279634ca207ab6bb3e73d2d7b2","container_hash":"gcr.io/cf-pks-releng-environments/busybox@sha256:679b1c1058c1f2dc59a3ee70eed986a88811c0205c8ceea57cec5f22d2c3fbb1","container_image":"gcr.io/cf-pks-releng-environments/busybox:latest"},"test2":"true","test3":"true","cluster_name":"aaa"}]
test3: [1640163220.340100423, {"stream":"stdout","logtag":"F","message":"[DEBUG] i=614","kubernetes":{"pod_name":"counter-1","namespace_name":"log-1","pod_id":"d76c604a-dc33-4ab6-b1b5-0f3bad72555d","host":"5c132468-b25d-4dbf-b71e-4a4aee46a011","container_name":"container1","docker_id":"f3be888c4a6d5bd71e1c00a7de63fe0bbc73df279634ca207ab6bb3e73d2d7b2","container_hash":"gcr.io/cf-pks-releng-environments/busybox@sha256:679b1c1058c1f2dc59a3ee70eed986a88811c0205c8ceea57cec5f22d2c3fbb1","container_image":"gcr.io/cf-pks-releng-environments/busybox:latest"},"test2":"true","test3":"true","cluster_name":"aaa"}]
```

--- a/plugins/filter_rewrite_tag/rewrite_tag.c
+++ b/plugins/filter_rewrite_tag/rewrite_tag.c
@@ -323,6 +323,9 @@ static int process_record(const char *tag, int tag_len, msgpack_object map,
 
     mk_list_foreach(head, &ctx->rules) {
         rule = mk_list_entry(head, struct rewrite_rule, _head);
+        if (rule) {
+            *keep = rule->keep_record;
+        }
         ret = flb_ra_regex_match(rule->ra_key, map, rule->regex, &result);
         if (ret < 0) { /* no match */
             rule = NULL;
@@ -359,7 +362,7 @@ static int process_record(const char *tag, int tag_len, msgpack_object map,
         return FLB_FALSE;
     }
 
-    *keep = rule->keep_record;
+
     return FLB_TRUE;
 }
 

